src/App.js

```diff
@@ -1,88 +1,188 @@
-  console.log('Deserializing decoding: %o', decoding);
-  switch (decoding.kind) {
-    case "function": {
-      return {
-        ...decoding,
-        class: Codec.Format.Utils.Serial.deserializeType(decoding.class),
-        arguments: decoding.arguments.map(({ name, value }) => ({
-          name,
-          value: Codec.Format.Utils.Serial.deserializeResult(value)
-        }))
-      };
-    }
-    case "constructor": {
-      return {
-        ...decoding,
-        class: Codec.Format.Utils.Serial.deserializeType(decoding.class),
-        arguments: decoding.arguments.map(({ name, value }) => ({
-          name,
-          value: Codec.Format.Utils.Serial.deserializeResult(value)
-        }))
-      };
+import React, { useState, useEffect } from "react";
+import { css } from "@emotion/react";
+import "./App.css";
+import EthTxParams from "./eth-tx-params";
+
+// import { Web3ReactProvider } from "@web3-react/core";
+// import { Web3Provider } from "@ethersproject/providers";
+// import { useWeb3React } from "@web3-react/core";
+
+// import decodings from "./decodings";
+import * as Codec from "@truffle/codec";
+import TextField from "@mui/material/TextField";
+import Button from "@mui/material/Button";
+import CircularProgress from "@mui/material/CircularProgress";
+
+import { txs, getDecoding } from "./generate-decodings";
+
+// const getLibrary = (provider: any): Web3Provider => {
+//   const library = new Web3Provider(provider);
+//   library.pollingInterval = 12000;
+//   return library;
+// };
+
+const App = () => {
+  // const { connector, library, chainId, account, activate, deactivate, active } =
+  //   useWeb3React();
+
+  const [template, setTemplate] = useState(-1);
+  const [txParams, setTxParams] = useState({});
+  const [loading, setLoading] = useState(false);
+
+  const [definitions, setDefintions] = useState({});
+  const [data, setData] = useState({});
+
+  // make sure we have a template selection
+  useEffect(() => {
+    if (template >= 0) {
+      setTxParams(txs[template]);
     }
-    case "message": {
-      return {
-        ...decoding,
-        class: Codec.Format.Utils.Serial.deserializeType(decoding.class)
-      };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [template]);
+
+  useEffect(() => {
+    (async () => {
+      if (template > -1 && txs[template]?.to) {
+        const txParams = txs[template];
+        setData({});
+        setDefintions({});
+        setLoading(true);
+
+        const { decoding, definitions } = await getDecoding(txParams);
+        setDefintions(definitions);
+
+        const data = deserializeCalldataDecoding(decoding);
+
+        setData(data);
+        setLoading(false);
+      }
+    })();
+  }, [template]);
+
+  const deserializeCalldataDecoding = (decoding) => {
+    switch (decoding.kind) {
+      case "function": {
+        return {
+          ...decoding,
+          class: Codec.Format.Utils.Serial.deserializeType(decoding.class),
```

Comment:
Note that `Codec.Format.Utils.Serial` only exists in `@truffle/codec@cheerios`... `latest` doesn't have this functionality, so we'll need to take this out as part of that switch (sadly we can't maintain `cheerios` moving forward)